quarkus/config-api/src/main/java/org/keycloak/config/OptionBuilder.java

```diff
@@ -128,7 +145,26 @@ public Option<T> build() {
             deprecated();
         }
 
-        return new Option<T>(type, key, category, hidden, build, description, defaultValue, expectedValues, deprecatedMetadata);
+        Class<?> expected = type;
+        if (auxiliaryType != null) {
+            expected = auxiliaryType;
+        }
+
+        if (CollectionUtil.isEmpty(expectedValues)) {
```

Comment:
We talked about adding to getExpectedValues javadoc a reference to strict method. There is also some subtle behavior here that should be mentioned - if you call builder.expectedValues() that doesn't mean no expected values, it means infer the default.